### ISSUE TITLE
How to cast an array of objects containing multiple fields depending on each other ?

### ISSUE THREAD

**Issue by dalawwa** [jquense/yup#1040]
Given an array of objects defined as
```
  {
	id: number,
	isRisky: boolean
  }
```

I would like to perform the following transformation
* when `id < 0`, `isRisky` should be transformed to `false`
* when `isRisky === true and id % 2 === 0`, `id` should be transformed to `-11`

To do so, I use cast with `context` to avoid running into `cyclic dependency` error

[Here](https://codesandbox.io/s/yup-interdependant-validation-flat-j6gvq?file=/src/App.tsx) is a working example of individual object transform

However, I am not sure how to work with a context array
[Attempt at validating in array](https://codesandbox.io/s/yup-interdependant-validation-array-lkpou?file=/src/App.tsx)

**TLDR**
[simplified runnable test](https://codesandbox.io/s/bold-yalow-j9xco?file=/src/index.test.js) not working
[almost work around](https://codesandbox.io/s/reverent-knuth-69l76) but how to access element index in the array dynamically (line 7) ?

Cheers

**Comment by jquense** [comment]
I think you are using `context` whe nyou don't need to. `context` if for referencing external data in a condition, you want to use internal values, e.g. the objects `id` field just do `.when('id', yourLogic)` and it'll work

**Comment by dalawwa** [comment]
Thank you for your quick answer and for this amazing library and community you've built.

I am not sure how I should do to avoid cyclic dependency issues without using context, as in my overall usecase both properties depend on each other.

**Comment by jquense** [comment]
search the issues for dealing with cyclic dependencies, there is an escape hatch. But as a warning, there are cyclical errors because it's generally not safe to have two fields transform each other, there is no way to decide which field to "do first" since they both change each other.

**Comment by dalawwa** [comment]
Ok, I will try with the escape hatch, I believe it is the second argument to `shape` (noSortEdges in the signature)
